Adaptors/Amqp/src/ConnectionAmqp.cs

```diff
@@ -37,50 +37,74 @@ namespace ArmoniK.Core.Adapters.Amqp;
 [UsedImplicitly]
 public class ConnectionAmqp : IConnectionAmqp
 {
-  private readonly AsyncLazy               connectionTask_;
-  private readonly ILogger<ConnectionAmqp> logger_;
-  private readonly QueueCommon.Amqp        options_;
-  private          bool                    isInitialized_;
+  private readonly ExecutionSingleizer<Connection> connectionSingleizer_;
+  private readonly ILogger<ConnectionAmqp>         logger_;
+  private readonly QueueCommon.Amqp                options_;
+  private          Connection?                     connection_;
 
   public ConnectionAmqp(QueueCommon.Amqp        options,
                         ILogger<ConnectionAmqp> logger)
   {
-    options_        = options;
-    logger_         = logger;
-    connectionTask_ = new AsyncLazy(() => InitTask(this));
+    options_              = options;
+    logger_               = logger;
+    connectionSingleizer_ = new ExecutionSingleizer<Connection>();
   }
 
-  public Connection? Connection { get; private set; }
-
   public Task<HealthCheckResult> Check(HealthCheckTag tag)
     => tag switch
        {
-         HealthCheckTag.Startup or HealthCheckTag.Readiness => Task.FromResult(isInitialized_
+         HealthCheckTag.Startup or HealthCheckTag.Readiness => Task.FromResult(connection_ is not null
                                                                                  ? HealthCheckResult.Healthy()
                                                                                  : HealthCheckResult.Unhealthy($"{nameof(ConnectionAmqp)} is not yet initialized.")),
-         HealthCheckTag.Liveness => Task.FromResult(isInitialized_ && Connection is not null && Connection.ConnectionState == ConnectionState.Opened
+         HealthCheckTag.Liveness => Task.FromResult(connection_ is not null && connection_.ConnectionState == ConnectionState.Opened
                                                       ? HealthCheckResult.Healthy()
                                                       : HealthCheckResult.Unhealthy($"{nameof(ConnectionAmqp)} not initialized or connection dropped.")),
          _ => throw new ArgumentOutOfRangeException(nameof(tag),
                                                     tag,
                                                     null),
        };
 
-  public async Task Init(CancellationToken cancellationToken = default)
-    => await connectionTask_;
+  public Task Init(CancellationToken cancellationToken = default)
+    => GetConnectionAsync(cancellationToken);
 
-  private static async Task InitTask(ConnectionAmqp    conn,
-                                     CancellationToken cancellationToken = default)
+  public async Task<Connection> GetConnectionAsync(CancellationToken cancellationToken = default)
```

Comment:
You can remove "async" if you return the Task from the ExecutionSingleizer directly.